src/main/kotlin/com/sourcegraph/cody/edit/sessions/FixupSession.kt

```diff
@@ -349,6 +354,31 @@ abstract class FixupSession(
           }
         }
 
+        // Skip the formatting if these edits are part of an undo operation.
+        if (!undoInProgress) {
+          // It is safe to use selectionRange here, because it is set in the constructor
+          // by calling triggerFixupAsync() which in turn calls ensureSelectionRange().
+          // The value of this property does not change
+          // when the user selects a different snippet in the UI.
+          // FIXME this causes a problem when the entire file becomes shorter than the original
+          // selection range.
+          selectionRange!!.let {
+            val tr = TextRange(it.start.toOffset(document), it.end.toOffset(document))
+            val input = document.getText(tr)
+            val formatted =
+                CodyFormatter.formatStringBasedOnDocument(
+                    input, project, document, tr, tr.startOffset)
+            if (input.replace(Regex("\\s"), "") != formatted.replace(Regex("\\s"), "")) {
```

Comment:
Do we need a regex there? isn't it the same as just calling `.trimStart()`?

Reply:
I want to normalise all whitespace in the string, not just the prefix. 

This is a sanity check to cancel formatting if the result is obviously wrong and applying it would break the code. However, I've already fixed the mistake that had caused such results, so this check is probably redundant by now.